extensions/telegram_bot/script.py

```diff
@@ -0,0 +1,430 @@
+            char_list = self.get_characters_json_list()
+            to_send = []
+            for i, char in enumerate(char_list):
+                to_send.append("/" + self.load_cmd + str(i) + " " + char.replace(".json", ""))
+                if i % 50 == 0 and i != 0:
+                    send_text = "\n".join(to_send)
+                    context.bot.send_message(text=send_text, chat_id=chat_id)
+                    to_send = []
+            if len(to_send) > 0:
+                send_text = "\n".join(to_send)
+                context.bot.send_message(text=send_text, chat_id=chat_id)
+
+    # =============================================================================
+    # answer generator
+    def generate_answer(self, user_in, chat_id):
+        # if generation will fail, return "fail" answer
+        answer = self.GENERATOR_FAIL
+        # acquire generator lock if we can
+        while self.generator_lock.locked():
+            sleep(1)
+        self.generator_lock.acquire(timeout=600)
+        try:
+            # Preprocessing user text:
+            # If notebook - append to history only user text;
+            self.users[chat_id]["user_in"].append(user_in)
+            if self.bot_mode == "notebook":
+                self.users[chat_id]["history"].append(user_in)
+            # If user_in starts with prefix - impersonate-like (useful if you try to get "environment/impersonate view")
+            elif user_in.startswith(self._impersonate_prefix):
+                # adding "" history line to prevent bug in history sequence, user_in is prefix for bot
+                self.users[chat_id]["history"].append("")
+                self.users[chat_id]["history"].append(user_in + ":")
+            # if user_in is "" - no user text, it is like continue generation
+            elif user_in == "":
+                # adding "" history line to prevent bug in history sequence
+                self.users[chat_id]["history"].append("")
+                self.users[chat_id]["history"].append(self.users[chat_id]["name2"] + ":")
+            # If not notebook then chat - add "name1&2:" to user and bot message (generation from name2 point of view);
+            else:
+                self.users[chat_id]["history"].append(self.users[chat_id]["name1"] + ":" + user_in)
+                self.users[chat_id]["history"].append(self.users[chat_id]["name2"] + ":")
+            # Set eos_token and stopping_strings.
+            stopping_strings = []
+            eos_token = None
+            if self.bot_mode in ["chat", "chat-restricted"]:
+                # don't know why, but better works without stopping_strings
+                # stopping_strings = [f'\n{self.users[chat_id]["name2"]}:', f'\n{self.users[chat_id]["name1"]}:']
+                eos_token = '\n'
+            # Make prompt
+            prompt = self.users[chat_id]["context"] + "\n" + "\n".join(self.users[chat_id]["history"])
+            # Generate!
+            generator = generate_reply(

```

Comment:
This is not working with the current main branch, use something like this instead:
```
        # Generate!
        params = {
            "max_new_tokens": 1024,
            "do_sample": True,
            "temperature": 0.72,
            "top_p": 0.73,
            "top_k": 0,
            "typical_p": 1,
            "repetition_penalty": 1.1,
            "encoder_repetition_penalty": 1,
            "min_length": 0,
            "no_repeat_ngram_size": 0,
            "num_beams": 1,
            "penalty_alpha": 0,
            "length_penalty": 1,
            "early_stopping": False,
            "seed": -1,
            "add_bos_token": True,
            "custom_stopping_strings": []
        }

        generator = generate_reply(
            question=prompt, state=params,
            eos_token=eos_token, stopping_strings=stopping_strings
        )
```

Reply:
Fixed.